edx/analytics/tasks/course_video.py

```diff
@@ -0,0 +1,667 @@
+"""
+Luigi tasks for extracting course video information from tracking logs.
+"""
+import math
+import html5lib
+import json
+from operator import itemgetter
+from datetime import datetime
+
+import luigi
+import luigi.hdfs
+import luigi.s3
+
+from edx.analytics.tasks.mapreduce import MapReduceJobTask, MapReduceJobTaskMixin
+from edx.analytics.tasks.pathutil import PathSetTask
+from edx.analytics.tasks.url import get_target_from_url, url_path_join
+from edx.analytics.tasks.mysql_load import MysqlInsertTask
+import edx.analytics.tasks.util.eventlog as eventlog
+import edx.analytics.tasks.util.opaque_key_util as opaque_key_util
+
+import logging
+log = logging.getLogger(__name__)
+
+################################
+# Task Map-Reduce definitions
+################################
+
+EVENT_ACTION_MAPPING = {
+    'play_video': 'P',
+    'seek_video': 'E',
+    'stop_video': 'S',
+    'load_video': 'L',
+}
+
+SEEK_INTERVAL_IN_SEC = 10
```

Comment:
the new endpoint has been exposed through https://github.com/ataki/edx-analytics-data-api/tree/course_video_api